apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,318 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
+  alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.Api.Messages
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Position
+
+  import Lexical.Test.Fixtures
+  import Lexical.Test.CodeSigil
+  import Lexical.Test.CursorSupport
+  import Messages
+
+  use ExUnit.Case, async: false
+
+  defp with_navigation_project(_ctx) do
+    project = project(:navigations)
+    {:ok, _, _} = RemoteControl.start_link(project, self())
+
+    on_exit(fn ->
+      :ok = RemoteControl.stop(project)
+    end)
+
+    RemoteControl.Api.schedule_compile(project, true)
+    assert_receive project_compiled(), 5000
+
+    %{project: project}
+  end
+
+  defp with_referenced_file(%{project: project}, file \\ "my_definition.ex") do
+    path = file_path(project, Path.join("lib", file))
+    %{uri: SourceFile.Path.ensure_uri(path)}
+  end
+
+  defp uses_file_uri(project) do
+    project
+    |> file_path(Path.join("lib", "my_module.ex"))
+    |> SourceFile.Path.ensure_uri()
+  end
+
+  defp open_uses_file(project, content) do
+    uri = uses_file_uri(project)
+
+    with :ok <- RemoteControl.call(project, SourceFile.Store, :open, [uri, content, 1]) do
+      RemoteControl.call(project, SourceFile.Store, :fetch, [uri])
+    end
+  end
+
+  setup_all ~w(with_navigation_project)a
```

Comment:
i think i'd prefer a normal `setup_all` block, it's clearer, and you're not reusing this. Also, using `~w` in this instance doesn't save anything compared to `[:with_navigation_project]`